### PR TITLE
feat(mobile): protected routing bb-29

### PR DESCRIPTION
**Pull request by ToyLess78**
Description:
This PR addresses the issue 29, which involves configuring navigation to ensure that users who are not signed in are redirected from protected screens to the Sign In page.

Changes Introduced:
Protected Routing Logic: Implemented logic to check user authentication status before allowing access to protected screens.
Navigation Configuration: Updated the navigation flow to redirect non-authenticated users to the Sign In page if they attempt to access a protected screen.

### CODE REVIEW

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { View } from "react-native";
+
+import { Text } from "~/libs/components/components";
```
Comment (revenets) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:4]:
```suggestion

import { Text, View } from "~/libs/components/components";
```

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/protected-route/protected-route.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import { RootScreenName } from "~/libs/enums/enums";
+import { useEffect, useNavigation } from "~/libs/hooks/hooks";
+import { type RootNavigationParameterList } from "~/libs/types/types";
+
+type Properties = {
+	children: ReactNode;
+};
+
+const ProtectedRoute: React.FC<Properties> = ({ children }) => {
```
Comment (revenets) [inline apps/mobile/src/libs/components/protected-route/protected-route.tsx:12]: There is no need to create a separate component for it. You should just maybe group some existing screens and render them conditionally depending on whether the user is authorized or not.

File: apps/mobile/src/navigations/root/root.tsx [hunk 3]
```diff
@@ -15,10 +16,24 @@ const screenOptions: NativeStackNavigationOptions = {
...
 		<NativeStack.Navigator screenOptions={screenOptions}>
-			<NativeStack.Screen component={Auth} name={RootScreenName.SIGN_IN} />
-			<NativeStack.Screen component={Auth} name={RootScreenName.SIGN_UP} />
+			{hasUser ? (
+				<NativeStack.Screen
+					component={WelcomeScreen}
+					name={RootScreenName.WELCOME}
+				/>
+			) : (
+				<>
```
Comment (revenets) [inline apps/mobile/src/navigations/root/root.tsx:32]:
```suggestion
				<NavigateStack.Group>
```

File: apps/mobile/src/screens/auth/auth.tsx [hunk 4]
```diff
@@ -22,10 +22,8 @@ import { SignInForm, SignUpForm } from "./components/components";
 const Auth: React.FC = () => {
 	const { name } = useAppRoute();
 	const dispatch = useAppDispatch();
-	const { dataStatus, user } = useAppSelector(({ auth }) => ({
-		dataStatus: auth.dataStatus,
-		user: auth.user,
-	}));
+	const dataStatus = useAppSelector((state) => state.auth.dataStatus);
+	const user = useAppSelector((state) => state.auth.user);
```
Comment (revenets) [inline apps/mobile/src/screens/auth/auth.tsx:26]:
```suggestion
	const { dataStatus, user } = useAppSelector((state) => state.auth);
```

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+import { Text, View } from "~/libs/components/components";
+
+const WelcomeScreen: React.FC = () => {
```
Comment (fshabanov) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:5]: Maybe Root? Then we will be able to use it in future

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx (hunk 5 above)
Comment (revenets) [reply apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:5]: Not sure that it should be Root. It's just a one of other screens which will be available for authenticated user.

**Review by revenets — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the main branch into yours

**Pull request merged**